pkg/policy/l4.go

```diff
@@ -583,6 +583,19 @@ func (l4Filter *L4Filter) toMapState(p *EndpointPolicy, identities Identities, f
 			keyToAdd.Identity = id.Uint32()
 			if entryCb(keyToAdd, &entry) {
 				p.PolicyMapState.denyPreferredInsertWithChanges(keyToAdd, entry, identities, features, changes)
+				// If Cilium is in dual-stack mode then the "World" identity
+				// needs to be split into two identities to represent World
+				// IPv6 and IPv4 traffic distinctly from one another.
+				if id == identity.ReservedIdentityWorld && option.Config.IsDualStack() {
```

Comment:
Why doesn't `cs.GetSelections()` already select both of the world identities if world was selected?

Reply:
^ Answer is that the `toEntities: - world` selector doesn't select the world-ipv4 / world-ipv6 identities currently, see my previous comments above. I'd argue that we should just put those labels into the world selector, then this code doesn't need to be handled as a special case here in `toMapState()`.

Reply:
I prefer to do that in another PR. Making these labels selectable **could** open up a can of worms.

Reply:
Done: https://github.com/cilium/cilium/issues/27494

Reply:
We did miss this logic; world is a label, not an identity. https://github.com/cilium/cilium/pull/29958 fixes this.

We might be able to remove this code now.